src/Processors/Sources/RemoteSource.cpp

```diff
@@ -51,12 +65,25 @@ Chunk RemoteSource::generate()
 
         was_query_sent = true;
     }
+#if defined(OS_LINUX)
+    auto res = query_executor->read(read_context);
+    if (std::holds_alternative<int>(res))
```

Comment:
Can we make it always return Block on non-Linux platforms, and do away with all other `ifdefs`? Non-linux platforms are secondary to us, so we might easily introduce bugs that we won't notice, if the code diverges too much.

Reply:
We can do it in this concrete place.
However, it is impossible to avoid `ifdefs` at all

Reply:
Removed define here.